### PR TITLE
Documentation for Boolean expressions

### PR DESCRIPTION
**Pull request by arj119**
Clubhouse ticket:
[ch152](https://app.clubhouse.io/manim-dsl/story/152/documentation-for-boolean-expressions)

Added documentation for boolean expressions and if statements

### CODE REVIEW

File: source/tutorial.rst [hunk 1]
```diff
@@ -23,36 +23,158 @@ Note that depending on your code style during declaration you may choose to omit
...
-  
+
+
+Boolean Expressions
+^^^^^^^^^^^^^^^^^^^
+Boolean expressions for many are quite familiar, as we have taken inspiration from other programming languages.
+
+**Equality**
+
+We have for equality two binary operators: equals ``==``, not equals ``!=``.
```
Comment (rbanerjee20) [inline source/tutorial.rst]: Can you rephrase like below: "For equality we have two binary operators"

File: source/tutorial.rst [hunk 2]
```diff
@@ -23,36 +23,158 @@ Note that depending on your code style during declaration you may choose to omit
...
+
+.. code :: javascript
     
+    let x = 5;
+    let y = x < 3;  // y = false
+    let z = x > 4;  // z = true
+    let a = x <= 5; // a = true
+    let b = x >= 6; // b = false
+
+**Logical And and Or**
```
Comment (rbanerjee20) [inline source/tutorial.rst]: Maybe rename this section header to Logical Operators - the "And and" is less readable

File: source/tutorial.rst [hunk 3]
```diff
@@ -23,36 +23,158 @@ Note that depending on your code style during declaration you may choose to omit
...
+
+These have been implemented with the following binary operators: logical and ``&&``, logical or ``||``.
+
+.. code :: javascript
+
+    let x = true;
+    let y = x && false; // y = false
+    let z = x || y;     // z = true
+
+**Logical Not**
```
Comment (rbanerjee20) [inline source/tutorial.rst]: Can include Logical Not under Logical Operators

File: source/tutorial.rst [hunk 4]
```diff
@@ -23,36 +23,158 @@ Note that depending on your code style during declaration you may choose to omit
...
+
+This is implemented as the standard ``!`` unary operator.
+
+..  code :: javascript
+
+    let x = true;
+    let y = !x;  // y = false
+
+
+**Precedence**
```
Comment (rbanerjee20) [inline source/tutorial.rst]: Maybe have this as a subheading under Logical Operators

File: source/tutorial.rst [hunk 5]
```diff
@@ -23,36 +23,158 @@ Note that depending on your code style during declaration you may choose to omit
...
+.. code :: javascript
     
     let stack = new Stack<number>;
 
 
 Control structures
 ^^^^^^^^^^^^^^^^^^
 
-``Work in progress!``
+**The if-then and if-then-else Statements**
```
Comment (rbanerjee20) [inline source/tutorial.rst:115]: Rename to Conditional Statements?

File: source/tutorial.rst [hunk 6]
```diff
@@ -23,36 +23,158 @@ Note that depending on your code style during declaration you may choose to omit
...
+        x = 10;
+    } else if(x < 12) {
+        x = 15;
+    } else {
+        x = 20;
+    }
+
+    let y = x;
+
+In the above example first the ``x < 4`` condition will be evaluated to false, then the ``x < 8`` which is also false and then the ``x < 12`` which evaluates to true. The program
```
Comment (rbanerjee20) [inline source/tutorial.rst]: "Then ``x < 8`` will also evaluate to false"? Keeps it consistent

File: source/tutorial.rst (hunk 5 above)
Comment (arj119) [reply source/tutorial.rst:115]: You could argue that while and do while are also conditional so I think this is fine. It is how it is in the java documentation

Comment (rbanerjee20) [reply source/tutorial.rst:115]: Okay sure, we can maybe have a whole conditional statements section later with if and while as subsections

**Review by rbanerjee20 — APPROVED** [review]
